weave/integrations/notdiamond/custom_router.py

```diff
@@ -0,0 +1,174 @@
+from typing import Any, Dict, Optional, Tuple, Union
+
+import pandas as pd
+from notdiamond.toolkit.custom_router import CustomRouter
+
+import weave
+from weave.flow.eval import EvaluationResults
+
+
+@weave.op(
+    name="notdiamond.custom_router.train_router",
+)
+def train_router(
```

Comment:
This file is quite interesting. Happy to land as is, but in the near future (a few weeks), we will have a more "direct" eval API that allows users to write eval results in batch, outside of the framework loops